packages/design-carousel/src/index.tsx

```diff
@@ -12,19 +15,33 @@ import type { Design } from '@automattic/design-picker/src/types';
 type DesignCarouselProps = {
 	onPick: ( design: Design ) => void;
 	selectedDesignSlugs?: string[];
+	flow?: string | null | undefined;
```

Comment:
I have a general comment here: can we not move _all_ the logic from functions below into props that are specified by callers? This should simplify the component and avoid reaching back out into external logic. As things stand, there is a _lot_ of flow-specific logic here, and it feels like pushing some of the decision-making up to the calling component will be easier to maintain in future.
Essentially, I am thinking we could update the props as per the suggestion here.
```suggestion
	flow?: string | null | undefined;
	designFilter?: ( design: Design ) => boolean; // My syntax could be wrong here; this would also mean `selectedDesignSlugs` could probably be dropped
	displayOnlyMobileView?: boolean;
	carouselDesktopOptions?: MShotsOptions; // should be defaulted if not present
	carouselMobileOptions?: MShotsOptions; // should be defaulted if not present
```

Maybe we can tackle that in a follow-up PR, but I think it would be a much cleaner interface.

Reply:
@daledupreez - That makes sense and is definitely something I can change in a follow-up PR. If everything is working with the current PR on your end, I can merge this and start working on those changes.